coderd/inboxnotifications.go

```diff
@@ -0,0 +1,376 @@
+	ctx := r.Context()
+
+	var (
+		apikey              = httpmw.APIKey(r)
+		targetsParam        = r.URL.Query().Get("targets")
+		templatesParam      = r.URL.Query().Get("templates")
+		readStatusParam     = r.URL.Query().Get("read_status")
+		startingBeforeParam = r.URL.Query().Get("starting_before")
+	)
+
+	targets, templates, readStatus, err := convertInboxNotificationParameters(ctx, api.Logger, targetsParam, templatesParam, readStatusParam)
+	if err != nil {
+		httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+			Message: "Invalid query parameter.",
+			Detail:  err.Error(),
+		})
+		return
+	}
+
+	startingBefore := dbtime.Now()
+	if startingBeforeParam != "" {
+		lastNotifID, err := uuid.Parse(startingBeforeParam)
+		if err != nil {
+			httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+				Message: "Invalid starting before.",
+			})
+			return
+		}
+		lastNotif, err := api.Database.GetInboxNotificationByID(ctx, lastNotifID)
+		if err != nil {
+			httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+				Message: "Failed to get notification by id.",
+			})
+			return
+		}
+		startingBefore = lastNotif.CreatedAt
+	}
+
+	notifs, err := api.Database.GetFilteredInboxNotificationsByUserID(ctx, database.GetFilteredInboxNotificationsByUserIDParams{
+		UserID:       apikey.UserID,
+		Templates:    templates,
+		Targets:      targets,
+		ReadStatus:   database.InboxNotificationReadStatus(readStatus),
+		CreatedAtOpt: startingBefore,
+	})
+	if err != nil {
+		api.Logger.Error(ctx, "failed to get filtered inbox notifications", slog.Error(err))
+		httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+			Message: "Failed to get filtered inbox notifications.",
+		})
+		return
+	}
+
+	unreadCount, err := api.Database.CountUnreadInboxNotificationsByUserID(ctx, apikey.UserID)
+	if err != nil {
+		api.Logger.Error(ctx, "failed to count unread inbox notifications", slog.Error(err))
+		httpapi.Write(ctx, rw, http.StatusInternalServerError, codersdk.Response{
+			Message: "Failed to count unread inbox notifications.",
+		})
+		return
+	}
+
+	httpapi.Write(ctx, rw, http.StatusOK, codersdk.ListInboxNotificationsResponse{
+		Notifications: func() []codersdk.InboxNotification {
+			notificationsList := make([]codersdk.InboxNotification, 0, len(notifs))
+			for _, notification := range notifs {
+				notificationsList = append(notificationsList, convertInboxNotificationResponse(ctx, api.Logger, notification))
+			}
+			return notificationsList
+		}(),
+		UnreadCount: int(unreadCount),
+	})
+}
+
+// updateInboxNotificationReadStatus changes the read status of a notification.
+// @Summary Update read status of a notification
+// @ID update-read-status-of-a-notification
+// @Security CoderSessionToken
+// @Produce json
+// @Tags Notifications
+// @Param id path string true "id of the notification"
+// @Success 201 {object} codersdk.Response
```

Comment:
Nit: `201` is used when resources are created: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201
`200 OK` is more acceptable here.